lib/engine/step/g_1860/track.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative 'tracker'
+
+module Engine
+  module Step
+    module G1860
+      class Track < Base
+        include Tracker
+        ACTIONS = %w[lay_tile pass].freeze
+
+        def actions(entity)
+          return [] if entity.company? || !can_lay_tile?(entity)
+
+          entity == current_entity ? ACTIONS : []
+        end
+
+        def description
+          'Lay Track'
+        end
+
+        def pass_description
+          @acted ? 'Done (Track)' : 'Skip (Track)'
+        end
+
+        def process_lay_tile(action)
+          lay_tile_action(action)
+          pass! unless can_lay_tile?(action.entity)
+        end
+
+        def reachable_path?(entity, path)
+          max_distance = @game.biggest_train(entity)
+          return false if max_distance.zero?
+
+          path_distances = @game.path_distances(entity)
+          path_distances[path] <= max_distance
+        end
+
+        def reachable_node?(entity, node)
+          max_distance = @game.biggest_train(entity)
+          return false if max_distance.zero?
+
+          node_distances = @game.node_distances(entity)
+          return false unless node_distances[node]
+
+          if node.city?
+            node_distances[node] < max_distance
+          else
+            node_distances[node] <= max_distance
+          end
+        end
+
+        def reachable_hex?(entity, hex)
+          max_distance = @game.biggest_train(entity)
```

Comment:
this pattern is repeated multiple times can probably be dryed up

Reply:
Yeah, I wasn't happy about that. Not sure what you mean by "dryed up", though.

Reply:
OK, refactored so that max_distances is only calculated once across most of these methods.